src/functions/vacation-request/list-vacation-request/handler.ts

```diff
@@ -6,22 +6,32 @@ import { middyfy } from "src/libs/lambda";
 /**
  * Labmda for listing all vacation requests from DynamoDB.
  */
-const listVacationRequestHandler: APIGatewayProxyHandler = async () => {
+const listVacationRequestHandler: APIGatewayProxyHandler = async (event) => {
+  const userId = event.queryStringParameters?.userId;
   try {
+    if (userId) {
```

Comment:
If userId can be undefined in the service as mentioned in the previous comment then we wouldn't need the separate requests here. It might be that filterExpression can't be undefined in which case you can't make these changes which is fine.

I think it would be best (if this structure needs to remain) if you declared const vacationRequests: VacationRequestModel[] = []
And then make the conditional request, update the declared vacationRequests and then return in the same return